app/src/androidTest/java/ch/epfl/sdp/MainActivityTest.java

```diff
@@ -25,6 +27,11 @@
     public final ActivityTestRule<MainActivity> mActivityRule =
             new ActivityTestRule<>(MainActivity.class);
 
+    @Before
+    public void setUp() {
+        initSafeTest(mActivityRule, true);
```

Comment:
PLEASE : ADD a @After public void release(){mActivityRule.release()}
Look at the method initSafeTest : it is there because we do not make those releases. If you do starts the activity, you should make a mActivityRule.release() after :)

Reply:
You're right, thanks for pointing this out